sys/include/net/nanocoap_sock.h

```diff
@@ -531,6 +531,20 @@ ssize_t nanocoap_sock_fetch_non(nanocoap_sock_t *sock, const char *path,
                                 const void *request, size_t len,
                                 void *response, size_t len_max);
 
+/**
+ * @brief   Simple non-confirmable GET
+ *
+ * @param[in]   sock    socket to use for the request
+ * @param[in]   path    remote path and query
+ * @param[out]  response buffer for the response, may be NULL
+ * @param[in]   len_max length of @p response
+ *
+ * @returns     length of response payload on success
+ * @returns     <0 on error
```

Comment:
What exactly is returned on error? (obviously not your fault here, `nanocoap_sock_get()` also does not specify this.)

Reply:
Additionally, why not sort it up to said function?

Reply:
> Additionally, why not sort it up to said function?

All the `_non`  variants are further down.

Reply:
Are you really sure? :sweat_smile: 

Isn't it: 
```
nanocoap_sock_Method()
nanocoap_sock_Method_non()
nanocoap_sock_Method_url()
```
With that, I would expect `.._get_non()` to be right after `.._get()`, near L390.

Reply:
(I wouldn't block the PR for that but it really bugs me :bug:)

Reply:
Ah only in the header you mean. You got the pattern 👍